lib/order.rb

```diff
@@ -0,0 +1,76 @@
+class Order
+
+  attr_accessor :complete
+
+  def initialize (menu)
+    @menu = menu.view
+    @order = {}
+    @total = 0.0
+    @complete = false
+
+  end
+
+  def view_order
+    @order.clone
+
+  end
+
+  def view_total
+    copy = @total.to_s
+    copy.to_f
+
+  end
+
+  def add pizza, number
+    raise 'order has already been completed' if complete
+    raise 'item not on menu' if !@menu.include?(pizza)
+    raise 'invalid number' if number % 1 != 0
+
+    if pizza_already_selected? pizza
+      @order[pizza] = updated_values_for_add(pizza, number)
+    else
+      raise 'invalid number' if number <= 0
+      temp_total = @menu[pizza]*number
+      @order[pizza] = [number, temp_total]
+    end
+
+    remove_zero_quantity_pizzas
```

Comment:
This would be a lot simpler if you had used an array instead of a hash BUT if you are going to do methods that deliberately alter state, and the return value is less important, use a bang (!) at the end
```ruby
remove_zero_quantity_pizzas!
```